React/props-state.md

```diff
@@ -0,0 +1,99 @@
+# `props` & `state`
+
+## Learning Goals
+- Examine how React allow components to manage data
+- Use `props` between components
+- Use `state` within a component
+- Impacts of both to component rendering
+
+## Overview
+Props & State are the two ways that React components manage and share their data.
+
+## `props`
+Props are **passed from the outside of the component** into the component itself. Props are accessible using `this.props` within a component. Props cannot be changed from within the component who received the prop values.
+
+#### Overview
+- Can be assigned to any variable name you want
+- Can contain any object or function you would like
+
+#### Examples
+**Syntax**
+
+Component A's `render` function contains the following JSX:
+`<ComponentB varName='value goes here'/>`
+
+From within Component B, you can now access this data using the `this.props.varName` variable.
+
+**More Complex**
+
+To add code to the value of the variable, use the JSX syntax for including code with the `{}`.
+
+A fairly common thing to do is to utilize `props` to pass an event handler from a parent component to a child component. In the parent component, we could pass this function within a prop as follows:
+`<ComponentB onChange={this.handleChange}`.
+
+Then, within the child component, whenever that event needs to be triggered you could use `this.props.onChange`. *Note:* it is the name of the variable not the name of the function from the parent component.
+
+## `state`
+State is managed **within a given component**. State can be accessed using `this.state` and it can be modified using `this.setState()`.
+
+#### Overview
+- Can be assigned to any object key you want
+- `setState` will always take an object as the parameter
+- `setState` will merge any params passed in with the existing state data, it won't overwrite it
+- We often set the initial state in the constructor function
+
+A really important aspect of state is that whenever the state of a component is changed, it will be **updated**, calling the `render` function. For now, you can think of `setState` as doing the `setState` operation PLUS the `render` _automatically_. We'll learn even more about what else our component does when it is updated later on.
```

Comment:
Are you referring to an instance variable in Ruby? I feel like they're not the same because they're not directly accessible or changeable in the way that instance variables are either. State should never be directly accessed externally